### PR TITLE
PLAT-754 Optimized Code Validation

### PR DESCRIPTION
**Pull request by oliver-richers**
now down to 7 seconds:
image
verbose:
image

The validation in EAS now also much faster:
image
Before it was much slower and some tests were even skipped:
image

### CODE REVIEW

File: platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/configuration/DbTableConfigurationCodeValidator.java [hunk 1]
```diff
@@ -57,35 +60,53 @@ private DbTestDatabase createTestDatabase() {
...
+		var tables = new ArrayList<IDbTable<?, ?>>();
+		for (var root: JavaClasspath.getInstance().getPayloadRoots()) {
+			for (var javaClass: root.getAnalyzedClasses()) {
+				findTable(javaClass).ifPresent(tables::add);
+			}
+		}
+		return tables;
+	}
+
+	private Optional<IDbTable<?, ?>> findTable(AnalyzedJavaClass javaClass) {
```
Comment (alexander-schmidt) [inline platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/configuration/DbTableConfigurationCodeValidator.java:74]: this method is partly redundant to `DbTableFinder`, e.g. it searches for `TABLE_FIELD_NAME`, etc. ... can we somehow unify this?

**Review by alexander-schmidt — CHANGES_REQUESTED** [review]
We should also ensure that this works as intended with payload projects.

File: platform-common/src/main/java/com/softicar/platform/common/code/reference/point/SourceCodeReferencePointsLoader.java [hunk 2]
```diff
@@ -28,23 +29,30 @@ public SourceCodeReferencePointsLoader() {
...
+				.forEach(this::addReferencePoint);
 			this.loaded = true;
 		}
 
 		exceptionsCollector.throwExceptionIfNotEmpty();
 		return referencePoints;
 	}
 
-	private void loadReferencePoint(JavaClassName className) {
+	private boolean isPublic(Class<?> referencePointClass) {
```
[the remark this answers is not in the document]
Comment (oliver-richers) [reply platform-common/src/main/java/com/softicar/platform/common/code/reference/point/SourceCodeReferencePointsLoader.java:45]: did so

File: platform-common/src/main/java/com/softicar/platform/common/io/resource/supplier/container/validator/ResourceSupplierContainerResourceKeyUnambiguityValidator.java [hunk 3]
```diff
@@ -3,51 +3,41 @@
...
 import com.softicar.platform.common.io.resource.supplier.IResourceSupplier;
 import com.softicar.platform.common.io.resource.supplier.container.validator.error.ResourceSupplierContainerResourceKeyAmbiguousError;
 import java.util.Collection;
-import java.util.List;
 import java.util.Objects;
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
-class ResourceSupplierContainerAllFieldsValidator implements IConstantContainerValidator<IResourceSupplier> {
+class ResourceSupplierContainerResourceKeyUnambiguityValidator implements IConstantContainerValidator<IResourceSupplier> {
```
[the remark this answers is not in the document]
Comment (oliver-richers) [reply platform-common/src/main/java/com/softicar/platform/common/io/resource/supplier/container/validator/ResourceSupplierContainerResourceKeyUnambiguityValidator.java:13]: as you wish :)

File: platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/configuration/DbTableConfigurationCodeValidator.java (hunk 1 above)
Comment (oliver-richers) [reply platform-db-runtime/src/main/java/com/softicar/platform/db/runtime/table/configuration/DbTableConfigurationCodeValidator.java]: did so, saved some lines of code

**Review by alexander-schmidt — APPROVED** [review]

**Pull request merged**